src/main/java/uk/ac/cam/cl/dtg/isaac/api/AssignmentFacade.java

```diff
@@ -1197,4 +1148,20 @@ public Response deleteAssignment(@Context final HttpServletRequest request,
             return new SegueErrorResponse(Status.INTERNAL_SERVER_ERROR, "Unknown database error.").toResponse();
         }
     }
+
+    private static Map<RegisteredUserDTO, Map<String, Integer>> getUserQuestionMap(
+            Map<RegisteredUserDTO, Map<String, Map<String, List<LightweightQuestionValidationResponse>>>> questionAttemptsForAllUsersOfInterest
+    ) {
+        return questionAttemptsForAllUsersOfInterest.entrySet().stream().collect(
```

Comment:
The duplication is gone and this code is more concise, but I still find it quite hard to read. Although it could be due to my lack of understanding of the business logic.